src/components/Machine/CrafterMachine/CrafterMachine.jsx

```diff
@@ -1,16 +1,106 @@
-import React from 'react'
-
+import React, { Component } from 'react'
+import Popper from '@material-ui/core/Popper'
+import Fade from '@material-ui/core/Fade'
+import Paper from '@material-ui/core/Paper'
+import { Button } from '@material-ui/core'
 import '../Machine.css'
-import { crafterMachine as machine, areEquals } from '../../Game/Game.constants'
+import './CrafterMachine.css'
+import { crafterMachine as machine, recipes } from '../../Game/Game.constants'
+
+class CrafterMachine extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      open: false,
+      recipeSelected: recipes[0],
```

Comment:
I think It is not a good idea to depend on global variables(DI pls).
A good practice for me would be to pass the `recipes` as props, so we would be able to change that parameter when we instantiate that component(in a test for example/mocking).

Reply:
mmmm i think that is more specify of this machines, also, why pass as props ? the recipes never change. Maybe, if we think and discuss a good way, it could be possible, yes.